From f75b2f6aba9f753e8b61ef3624b4aa39ff616f81 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 18 Jan 2022 16:11:34 +0200
Subject: [PATCH 20/64] s32: Use ARMv8 linker file

Remove S32 linker file as it's outdated and the workarounds added
there are outdated.

Issue: ALB-8334
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/u-boot.lds | 97 -------------------------------
 include/configs/s32.h             |  2 +-
 2 files changed, 1 insertion(+), 98 deletions(-)
 delete mode 100644 arch/arm/cpu/armv8/s32/u-boot.lds

diff --git a/arch/arm/cpu/armv8/s32/u-boot.lds b/arch/arm/cpu/armv8/s32/u-boot.lds
deleted file mode 100644
index 2ff60d1f61..0000000000
--- a/arch/arm/cpu/armv8/s32/u-boot.lds
+++ /dev/null
@@ -1,97 +0,0 @@
-/* SPDX-License-Identifier:     GPL-2.0+ */
-/*
- * (C) Copyright 2013
- * David Feng <fenghua@phytium.com.cn>
- *
- * (C) Copyright 2002
- * Gary Jennejohn, DENX Software Engineering, <garyj@denx.de>
- *
- * This file is based on the linker script file arch/arm/cpu/armv8/u-boot.lds
- * u-boot version v2016.01 and modified for alignment required by s32 platforms
- *
- * Copyright 2018 NXP
- */
-
-OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
-OUTPUT_ARCH(aarch64)
-ENTRY(_start)
-SECTIONS
-{
-	. = 0x00000000;
-
-	. = ALIGN(8);
-	.text :
-	{
-		*(.__image_copy_start)
-		CPUDIR/start.o (.text*)
-		*(.text*)
-	}
-
-	. = ALIGN(8);
-	.rodata : { *(SORT_BY_ALIGNMENT(SORT_BY_NAME(.rodata*))) }
-
-	. = ALIGN(32);
-	.data : {
-		*(.data*)
-	}
-
-	. = ALIGN(8);
-
-	. = .;
-
-	. = ALIGN(8);
-	.u_boot_list : {
-		KEEP(*(SORT(.u_boot_list*)));
-	}
-
-	. = ALIGN(8);
-
-	.image_copy_end :
-	{
-		*(.__image_copy_end)
-	}
-
-	. = ALIGN(8);
-
-	.rel_dyn_start :
-	{
-		*(.__rel_dyn_start)
-	}
-
-	.rela.dyn : {
-		*(.rela*)
-	}
-
-	.rel_dyn_end :
-	{
-		*(.__rel_dyn_end)
-	}
-
-	_end = .;
-
-	. = ALIGN(32);
-
-#ifdef CONFIG_OF_SEPARATE
-	. = . + 0x400;
-#endif
-
-	.bss_start : {
-		KEEP(*(.__bss_start));
-	}
-
-	.bss : {
-		*(.bss*)
-		 . = ALIGN(32);
-	}
-
-	.bss_end : {
-		KEEP(*(.__bss_end));
-	}
-
-	/DISCARD/ : { *(.dynsym) }
-	/DISCARD/ : { *(.dynstr*) }
-	/DISCARD/ : { *(.dynamic*) }
-	/DISCARD/ : { *(.plt*) }
-	/DISCARD/ : { *(.interp*) }
-	/DISCARD/ : { *(.gnu*) }
-}
diff --git a/include/configs/s32.h b/include/configs/s32.h
index c09594dbb9..99bf0cdefd 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -494,6 +494,6 @@
 
 #define CONFIG_SYS_PCI_64BIT
 
-#define CONFIG_SYS_LDSCRIPT  "arch/arm/cpu/armv8/s32/u-boot.lds"
+#define CONFIG_SYS_LDSCRIPT  "arch/arm/cpu/armv8/u-boot.lds"
 
 #endif /* __S32_H */
-- 
2.17.1

